catalog/spatial/registry/registry-identification-plugin/src/main/java/org/codice/ddf/registry/identification/RegistryIdPostIngestPlugin.java

```diff
@@ -0,0 +1,147 @@
+import org.opengis.filter.Filter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.CatalogFramework;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.Result;
+import ddf.catalog.filter.FilterBuilder;
+import ddf.catalog.operation.CreateResponse;
+import ddf.catalog.operation.DeleteResponse;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.operation.UpdateResponse;
+import ddf.catalog.operation.impl.QueryImpl;
+import ddf.catalog.operation.impl.QueryRequestImpl;
+import ddf.catalog.plugin.PluginExecutionException;
+import ddf.catalog.plugin.PostIngestPlugin;
+
+/**
+ * The RegistryIdPostIngestPlugin collects a set of registry-ids that are currently in the system
+ * and makes them available to the IdentificationPlugin for duplication checking.
+ *
+ * Initially the catalog is queried to initialize the set of registry-ids. After that the list is
+ * maintained by the process methods for create and delete responses.
+ */
+public class RegistryIdPostIngestPlugin implements PostIngestPlugin {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(IdentificationPlugin.class);
+
+    private static final int RETRY_INTERVAL = 30;
+
+    private ScheduledExecutorService executorService;
+
+    private CatalogFramework catalogFramework;
+
+    private FilterBuilder filterBuilder;
+
+    private Set<String> registryIds = ConcurrentHashMap.newKeySet();
+
+    public Set<String> getRegistryIds() {
+        return Collections.unmodifiableSet(registryIds);
+    }
+
+    @Override
+    public CreateResponse process(CreateResponse input) throws PluginExecutionException {
+        if (input != null && input.getCreatedMetacards() != null) {
+            input.getCreatedMetacards()
+                    .stream()
+                    .filter(RegistryUtility::isRegistryMetacard)
+                    .forEach(e -> registryIds.add(RegistryUtility.getRegistryId(e)));
+        }
+        return input;
+    }
+
+    @Override
+    public UpdateResponse process(UpdateResponse input) throws PluginExecutionException {
+        return input;
+    }
+
+    @Override
+    public DeleteResponse process(DeleteResponse input) throws PluginExecutionException {
+        if (input != null && input.getDeletedMetacards() != null) {
+            input.getDeletedMetacards()
+                    .stream()
+                    .filter(RegistryUtility::isRegistryMetacard)
+                    .forEach(e -> registryIds.remove(RegistryUtility.getRegistryId(e)));
+        }
+        return input;
+    }
+
+    public void init() {
+        try {
+            List<Metacard> registryMetacards;
+            Filter registryFilter = filterBuilder.attribute(Metacard.TAGS)
+                    .is()
+                    .like()
+                    .text(RegistryConstants.REGISTRY_TAG);
+            QueryImpl query = new QueryImpl(registryFilter);
```

Comment:
Used the same value we had in other places. 1000 is the maximum some sources can handle but that shouldn't matter here since we are dealing with the local catalog and locally created nodes. Basically we want all of them and we don't want to page through the results. In practice the most we have ever seen in a registry is ~30 and of those only a couple are locally created.  Let me know if you think we need to change this.